zerver/views/streams.py

```diff
@@ -607,16 +616,39 @@ def get_streams_backend(
     return json_success({"streams": streams})
 
 @has_request_variables
-def get_topics_backend(request: HttpRequest, user_profile: UserProfile,
-                       stream_id: int=REQ(converter=to_non_negative_int,
-                                          path_only=True)) -> HttpResponse:
-    (stream, recipient, sub) = access_stream_by_id(user_profile, stream_id)
+def get_topics_backend(
+        request: HttpRequest, maybe_user_profile: Union[UserProfile, AnonymousUser],
+        stream_id: int=REQ(converter=to_non_negative_int, path_only=True),
+        narrow: Optional[List[Dict[str, Any]]]=REQ('narrow', converter=narrow_parameter, default=None),
```

Comment:
Hmm, I don't think it makes sense to pass the `narrow` parameter here given that it's mostly discarded AND we have use cases in the webapp where we might call this endpoint without a clear narrow.  

I'm basically OK with having this endpoint be less explicit than `GET /messages` about requiring an explicit request to allow web-public access to web-public streams; it'll be used 1000x by hand-written clients that might have broken auth, and likely those clients would have caught it via `GET /messages` in the first place anyway.

Reply:
Okay, I felt weird adding a narrow argument there myself.

Reply:
I'd encourage you to flag things like this that you felt were questionable in the PR description (or a Zulip thread) as a decision deserving attention; that can save a bit of time as one can often answer that sort of question without reading much of the PR.

One technique I like to use is taking notes on items to follow-up on, either on paper or in a text file, while working on a project, and then having a habit of checking that list when submitting the PR.

Reply:
Okay!